File: 设计模式ProgrammingMode/adapter/adater/src/adapter_object.java
public class adapter_object {
    public static void main(String[] args){
        Target adapter = new Adapter();
        adapter.request();
    }
}

interface Target{
    public void request();
}

class Adaptee{
    public void specificRequest(){
        System.out.println("Adaptee specificRequest func is called");
    }
}

class Adapter extends Adaptee implements Target{
    private Adaptee adaptee = new Adaptee();
    public void request(){
        adaptee.specificRequest();
    }
}

File: 设计模式ProgrammingMode/adapter/adapter_object.cpp
//
// Created by Planck Chang on 2022/10/9.
//

#include <iostream>

using namespace std;

class Target{
public:
    virtual void request() = 0;
};

class Adaptee{
public:
    void specificRequest(){
       cout << "Adaptee specificRequest func is called\n";
    }
};

class Adapter:public Target{
public:
    void request(){
        adaptee.specificRequest();
    }
private:
    Adaptee adaptee;
};

int main(){
    Target * interface = new Adapter();
    interface->request();
}
